src/main/java/com/microsoft/azure/vmagent/AzureVMCloud.java

```diff
@@ -344,21 +344,27 @@ public int getApproximateVirtualMachineCount() {
 
     /**
      * Given the number of VMs that are desired, returns the number of VMs that
-     * can be allocated.
+     * can be allocated and adjusts the number of VMs we believe exist
      *
      * @param quantityDesired Number that are desired
-     * @return Number that can be allocated
+     * @return Number that can be allocated.
      */
-    public int getAvailableVirtualMachineCount(int quantityDesired) {
+    public int getAndAdjustVirtualMachineCounts(int quantityDesired) {
```

Comment:
With this method, seems there is no need to have `adjustVirtualMachineCount()`? (In theory every time you adjust the VM count, you need to make sure the adjusted number is within [0, maxVirtualMAchinesLimit].) Shall we consolidate these two methods into one?